lib/index.js

```diff
@@ -0,0 +1,437 @@
+                Object.assign(is, flattenEvents(member, [...path, member.id]));
+            });
+        }
+
+        return is;
+    },
+
+    /**
+     * A function that publish execution report to remote.
+     *
+     * @param {Object} newman - The collection run object, with a event handler setter,
+     * used to enable event wise reporting.
+     * @param {Object} options - The set of reporter run options.
+     * @param {Object} options.api - Remote endpoint to publish the report.
+     * @param {String} options.agentId - Remote agent identifier
+     * @param {String} options.agentName - Remote agent name
+     * @param {String} options.agentVersion - Remote agent version
+     * @param {Object} cr - The set of all the collection run options.
+     * @returns {*}
+     */
+    remoteReporter = function (newman, options, cr) {
+        if (!options.api) {
+            return console.error('newman-reporter-remote:\n  error: missing required option `api`');
+        }
+
+        try {
+            typeof options.api === 'string' && (options.api = JSON.parse(options.api));
+        }
+        catch (e) {
+            return console.error('newman-reporter-remote:\n  error: ', e);
+        }
+
+        const runId = uuid(),
+            collectionInfo = cr.collection.toJSON().info;
+
+        var collection = {
+                id: cr.collection.id,
+                $ref: ['collection', cr.collection.id],
+                info: {
+                    name: collectionInfo.name,
+                    schema: collectionInfo.schema
+                }
+            },
+            environment = {
+                id: cr.environment.id,
+                $ref: ['environment', cr.environment.id],
+                info: {
+                    name: cr.environment.name
+                }
+            },
+            run = {
+                id: runId,
+                type: 'waterfall',
+                agent: {
+                    id: options.agentId || uuid(),
+                    name: options.agentName || 'newman',
+                    version: options.agentVersion || cr.newmanVersion,
+                    platform: os.platform()
+                }
+            },
+            executions = [],
+            requests = {},
+            responses = {},
+            assertions = {},
+            errors = {},
+            events = {},
+            scripts = {},
+            items = {},
+            scriptStartTime,
+            iterationStartTime,
+            itemStartTime,
+            requestStartTime,
+            consoles = {},
+
+            _events = flattenEvents(cr.collection),
+            _items = flattenItems(cr.collection.items.members);
+
+        newman.on('beforeIteration', function () {
+            iterationStartTime = Date.now();
+        });
+
+        newman.on('iteration', function (err, data) {
+            if (err) {
+                !(err instanceof SerialisedError) && (err = new SerialisedError(err, true));
```

Comment:
Don't change the type of an entity on the fly. This leads to an optimization bailout (Link TBA)


```suggestion
            errors[err.id] = err instanceof SerialisedError ? err : new SerialisedError(err, true);
```